### DISCUSSION TITLE
504 gateway timeout when upload/delete plugins

### DISCUSSION THREAD

**Discussion by yongzhang** [docker-library/wordpress#593] [General]
image version: 5.7.0-php7.4
Container was deployed inside k8s, behind traefik ingress.

Issue:
When I upload a plugin zip file and install, the browser will show 504 gateway timeout after a while,
but after that, if I go back to the plugin list page and refresh the page, I can find the plugin was installed.
Then I delete the plugin, it will stuck at "deleting", but after refreshing the page, plugin disappeared.
I don't understand what's happening here, please help. Thanks.

**Comment by wglambert** [comment]
Is there anything in the `kubectl logs`?

**Reply by yongzhang** [reply]
I removed all my images by mistake :(
But the overall procedure is:
1. Upload a zip file and click install
2. The page is keep loading, then goes to a 504 page:
![image](https://user-images.githubusercontent.com/15604715/112586782-57a97180-8e37-11eb-89d5-4c7d88d441aa.png)
3. If I go back to plugin list page, I can find the plugin was there, I can also find the folder in wp-content/plugins directory.
4. Check the logs, nothing special:
![image](https://user-images.githubusercontent.com/15604715/112586852-7f003e80-8e37-11eb-9bb4-f95e24a38211.png)

**Reply by yongzhang** [reply]
I guess I might hit this:
https://wordpress.org/support/topic/error-wordpress-could-not-establish-a-secure-connection-to-wordpress-org/

but not sure, because I install plugins by uploading the zip file not from marketplace.

logs:
```
"log":"[26-Mar-2021 05:14:22 UTC] PHP Warning:  An unexpected error occurred. Something may be wrong with WordPress.org or this server’s configuration. If you continue to have problems, please try the support forums.） in /var/www/html/wp-includes/update.php on line 410\n","stream":"stderr","time":"2021-03-26T05:14:22.390707063Z"
```

**Comment by yongzhang** [comment]
Found the root cause, it was the WAF service in front of my wordpress server.
